biahub/cli/slurm.py

```diff
@@ -22,3 +30,136 @@ def wait_for_jobs_to_finish(jobs: list[submitit.Job]) -> None:
             pass  # as_completed polls every 10 seconds by default, so we don't need to do anything here
         except Exception as e:
             print(f"Job {job.job_id} failed with exception: {e}")
+
+
+def check_job_logs(
+    log_dir: Union[str, Path],
```

Comment:
Instead of `Union[str, Path]` I recommend either using use: `str | Path` or `PathLike` from `os`. `str | Path` is the recommended notation for Union types for Python 3.10+ (it's less to type).

`PathLike` is a nice protocol if you want a catch-all for everything that is well... "path like".